node/src/Pos/DB/Block.hs

```diff
@@ -486,19 +497,24 @@ putBi k v = rocksPutBi k v =<< getBlockIndexDB
 delete :: (MonadRealDB ctx m) => ByteString -> m ()
 delete k = rocksDelete k =<< getBlockIndexDB
 
-getData ::  (MonadIO m, MonadCatch m, Bi v) => FilePath -> m (Maybe v)
+getData ::  forall m v . (MonadIO m, MonadCatch m, Bi v) => FilePath -> m (Maybe v)
 getData fp = flip catch handle $ liftIO $
     either (\er -> throwM $ DBMalformed $
-             sformat ("Couldn't deserialize "%build%", reason: "%build) fp er) pure .
-    decodeFull <$>
+             sformat ("Couldn't deserialize "%build%", reason: "%build) fp er)
+           checkDBVersion .
+    decodeFull @(Word8, v) <$>
     BS.readFile fp
   where
     handle e
         | isDoesNotExistError e = pure Nothing
         | otherwise = throwM e
+    checkDBVersion :: (Word8, v) -> Maybe v
+    checkDBVersion (dbVer, val)
+        | dbVer == dbSerializeVersion = pure val
+        | otherwise = throwM $ DBUnexpectedVersionTag dbSerializeVersion dbVer
 
 putData ::  (MonadIO m, Bi v) => FilePath -> v -> m ()
-putData fp = liftIO . BS.writeFile fp . serialize'
+putData fp = liftIO . BS.writeFile fp . dbSerialize
```

Comment:
According to [this article](https://github.com/input-output-hk/cardano-sl-articles/blob/master/db-migration-mainnet.md) values stored in files shouldn't be versioned. It means that `getData` shouldn't check version, for example, and should decode` v`, not `(Word8, v)`.

Reply:
Also PureDB implementation shouldn't do this thing. I guess this file shouldn't change at all (maybe something small only).

Reply:
@gromakovsky is it really a problem? Leaving this file untouched makes block properties fail, which suggests this module depends on DB functions that serialize using `dbSerializeValue`. Since we must either change all of them or none, and this issue is about _performing_ the change, I'd say we need to change this too.

Reply:
> Leaving this file untouched makes block properties fail,

It should be investigate then.

> is it really a problem?

Are we talking about pure DB or real DB?
Regarding real: according to the spec values on disk should be serialized as is, without versions. I wasn't designing this spec, so I can't say how critical it is. If you really want to use prefixes, you can ask @georgeee about it.
Regarding pure: I think that the pure one should use exactly the same format as the real one.

> Since we must either change all of them or none, and this issue is about performing the change

This issue is about changing format in RocksDB. For blocks we are using plain files. It wasn't clear from the issue description whether we need versioning for blocks, but @georgeee gave me a link to that article which explicitly says that we don't need it for data stored in files.